tools/cldr-code/src/test/java/org/unicode/cldr/unittest/TestExampleGenerator.java

```diff
@@ -1645,4 +1662,140 @@ public void TestUnicodeSetExamples() {
+            return ComparisonChain.start()
+                    .compare(sectionId, o.sectionId)
+                    .compare(pageId, o.pageId)
+                    .compare(starred, o.starred)
+                    .result();
+        }
+
+        @Override
+        public boolean equals(Object obj) {
+            return compareTo((MissingKey) obj) == 0;
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hashCode(sectionId, pageId, starred);
+        }
+    }
+
+    static final Multimap<String, String> OK_IF_MISSING =
+            ImmutableMultimap.<String, String>builder()
+                    .put("//ldml/characters/moreInformation", "")
+                    // mul➔«Multiple languages»; zxx➔«No linguistic content»
+                    .putAll(
+                            "//ldml/localeDisplayNames/languages/language[@type=\"*\"]",
+                            "mul",
+                            "zxx")
+                    .build();
+
+    public void TestForMissing() {
+        Factory factory = info.getCldrFactory(); // don't worry about examples for annotations
+        for (String localeId : List.of("en", "de")) {
+            CLDRFile cldrFile = factory.make(localeId, true);
+            ExampleGenerator exampleGenerator = new ExampleGenerator(cldrFile, info.getEnglish());
+            PathStarrer ps = new PathStarrer();
+            ps.setSubstitutionPattern("*");
+            Counter<MissingKey> countWithExamples = new Counter<>();
+            Map<MissingKey, String> samplesForWith = new HashMap<>();
+            Counter<MissingKey> countWithoutExamples = new Counter<>();
+            Multimap<MissingKey, String> samplesForWithout = TreeMultimap.create();
+            DtdData dtdData = DtdData.getInstance(DtdType.ldml);
+            PathHeader.Factory phf = PathHeader.getFactory();
+            final String separator = "•";
+
+            // collect all of the strings in the file to test
+            for (String xpath : cldrFile.fullIterable()) {
+                if (xpath.endsWith("/alias")) {
+                    continue;
+                }
+                final XPathParts parts = XPathParts.getFrozenInstance(xpath);
+                if (dtdData.isDeprecated(parts)) {
+                    continue;
+                }
+                Level level = SDI.getCoverageLevel(xpath, "en");
+                if (level.compareTo(Level.COMPREHENSIVE) == 0) {
+                    continue;
+                }
+                String value = cldrFile.getStringValue(xpath);
+                String starred = ps.set(xpath);
+                String attrs = ps.getAttributesString(separator);
+                if (OK_IF_MISSING.containsEntry(starred, attrs)) {
+                    logln("OK if missing: " + starred + ";\t" + attrs);
+                    continue;
+                }
+                String example = null;
+                PathHeader ph = phf.fromPath(xpath);
```

Comment:
could also check with CheckCLDR.Phase.getShowRowAction() to make sure it's actually survey tool visible.

Reply:
I added a check on that, but it didn't remove anything that wasn't already removed.